### PR TITLE
fix react warnings

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

fixes some of the warnings from emotion and react

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
Thank you much for doing this! @uladzimirdev 
Did you want to keep it within a certain scope or would you like to get rid of most (all) of the errors. Asking because I found more, but I am not sure do you want to keep iterating in this PR?

Maybe we can create a list somewhere and then addressing these errors separately?

File: frontend/src/metabase/components/Badge/Badge.styled.tsx [hunk 1]
```diff
@@ -43,7 +43,9 @@ export const MaybeLink = styled(RawMaybeLink)`
   }
 `;
 
-export const BadgeIcon = styled(Icon)<{ hasMargin: boolean }>`
+export const BadgeIcon = styled(Icon, {
+  shouldForwardProp: propName => propName !== "hasMargin",
```
Comment (npretto) [inline frontend/src/metabase/components/Badge/Badge.styled.tsx:47]: nit: I learned the other day that we have a util for this: https://github.com/metabase/metabase/blob/813258b9531e3cdac654864f4507aa8a27b4fb49/frontend/src/metabase/common/utils/doNotForwardProps.ts#L1-L3

**Review by romeovs — APPROVED** [review]
Other than @npretto's nit, this looks good to me.

File: frontend/src/metabase/components/Badge/Badge.styled.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/components/Badge/Badge.styled.tsx:47]: yup and I'd get rid of that util 😄

**Pull request merged**